### PR TITLE
Revert "Revert "MAINT: Remove Long and WidePanel (#15748)" (#15802)"

### PR DESCRIPTION
**Pull request by gfyoung**
With https://github.com/statsmodels/statsmodels/issues/3580 now resolved, we can safely revert #15802

### CODE REVIEW

**Comment by jreback** [comment]
going to merge #18333 which will causes this to fail. We weren't fully testing things in the downstream.

**Comment by jreback** [comment]
@gfyoung go ahead and rebase. This *should* fail. We can decide whether to xfail the test at a later point.

**Comment by gfyoung** [comment]
> Let's still wait until there is an actual statsmodels release. At this point we don't know yet who of statsmodels or pandas will release first.

@jorisvandenbossche : That's unfortunate, as they said that it would come out by end of summer.  Not sure what happened there.

> We can decide whether to xfail the test at a later point.

@jreback : Not sure what "the test" is, but yep, will do.

**Comment by gfyoung** [comment]
Ummm...@jreback : what test was supposed to fail? :smile:

**Comment by jorisvandenbossche** [comment]
> @jorisvandenbossche : That's unfortunate, as they said that it would come out by end of summer. Not sure what happened there.

Well, that's the reality of volunteer-based open source. It's indeed a bit annoying for us to have to keep it, but also not that costly. 
We can just leave the PR open until the new statsmodels is out I would say.

**Comment by gfyoung** [comment]
> Well, that's the reality of volunteer-based open source. It's indeed a bit annoying for us to have to keep it, but also not that costly.

Yeah...things happen.  I have no problems waiting.

**Comment by bashtage** [comment]
> Let's still wait until there is an actual statsmodels release. At this point we don't know yet who of statsmodels or pandas will release first.

IMO pandas should make this change since it provides incentive for statsmodels to drop a release.

**Comment by jreback** [comment]
@gfyoung once I merge this, it should fail: https://github.com/pandas-dev/pandas/pull/18362

turns out was skipping the tests :<

**Comment by jreback** [comment]
Then I agree with @bashtage we should xfail this test and merge. but let it fail to prove first.

**Comment by gfyoung** [comment]
> IMO pandas should make this change since it provides incentive for statsmodels to drop a release.

@bashtage : I actually hadn't thought of that.  Also makes sense.

**Comment by jreback** [comment]
@gfyoung rebase and let's see

**Comment by gfyoung** [comment]
@jreback : Can confirm that Travis is failing as a result of your PR.  I will `@xfail` the test once Circle and AppVeyor have finished.

**Comment by gfyoung** [comment]
@jreback : And adding `@xfail` does fix everything.  All is green again.

File: pandas/tests/test_panel.py [hunk 1]
```diff
@@ -2155,8 +2147,8 @@ def test_multiindex_get(self):
             assert (f1.items == [1, 2]).all()
             assert (f2.items == [1, 2]).all()
 
-            ind = MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1)],
```
Comment (jreback) [inline pandas/tests/test_panel.py]: what does this last section compare?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_panel.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tests/test_panel.py]: Not entirely sure?  I just saw this variable being assigned and not being used afterwards.

**Comment by jreback** [comment]
@TomAugspurger is there a statsmodel 0.9.0 release issue? (where we can reference this)

File: pandas/tests/test_panel.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/test_panel.py]: ok, if you can remove that (doesn't look like it does anything) would be great.

**Comment by gfyoung** [comment]
@jreback : Judging from the milestone search <a href="https://github.com/statsmodels/statsmodels/milestone/10">here</a>, <a href="https://github.com/statsmodels/statsmodels/issues/3431>statsmodels/statsmodels#3431</a> might be it?  A little sparse though.

File: pandas/tests/test_panel.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tests/test_panel.py]: Sure thing.  Done.

**Comment by TomAugspurger** [comment]
I don't think so.

**Comment by jorisvandenbossche** [comment]
Still -1 on merging this. I would say let's try to convince them with words to do a release instead of by breaking their code.

**Comment by jreback** [comment]
ok

@gfyoung let's remove the xfail and just leave this PR until/when statsmodels can do a release.

**Comment by gfyoung** [comment]
Alright.  I removed the `xfail` for the time being.  It might be awhile on this one, as it looks like a lot of stuff is still left for them to do for `0.9`.

**Comment by bashtage** [comment]
SM only has to release before 0.22.  I suspect this won't be out until med Q1 2018.  This should be plenty of time.

File: doc/source/whatsnew/v0.23.0.txt [hunk 2]
```diff
@@ -1079,6 +1079,8 @@ Removal of prior version deprecations/changes
   so operations would continue to work. This is now fully removed, so a ``Resampler`` will no longer forward compat operations (:issue:`20554`)
 - Remove long deprecated ``axis=None`` parameter from ``.replace()`` (:issue:`20271`)
 
+- The ``LongPanel`` and ``WidePanel`` classes have been removed (:issue:`10892`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: move to 0.24

**Review by jreback — CHANGES_REQUESTED** [review]
@gfyoung I think can finally merge this as sm 0.9.0 is out.

File: doc/source/whatsnew/v0.23.0.txt (hunk 2 above)
Comment (gfyoung) [reply doc/source/whatsnew/v0.23.0.txt]: Yep, will do.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Comment by gfyoung** [comment]
Sure thing.  Hopefully we won't need to revert this revert 😄

**Pull request merged**